pkg/service/cluster/service.go

```diff
@@ -667,10 +669,27 @@ func (s *Service) ListNodes(ctx context.Context, clusterID uuid.UUID) ([]Node, e
 			if err != nil {
 				s.logger.Error(ctx, "Failed to get number of shards", "error", err)
 			}
+			ni, err := client.NodeInfo(ctx, h)
+			if err != nil {
+				return nil, errors.Wrapf(err, "node info call of %s", h)
+			}
+			promAddr := ni.PrometheusAddress
+			if temp, err := netip.ParseAddr(ni.PrometheusAddress); err == nil {
+				promAddr = temp.String()
+			}
```

Comment:
Why is this step important?

Reply:
ni.PrometheusAddress is a string, so I assume in the code that it may hold everything